src/Domain/DomainHelpers.hpp

```diff
@@ -0,0 +1,329 @@
+ *     was determined by the Global CNS, and the location of the ids in the
+ * vector, which was determined by the Local CNS.
+ *     In fact, the array index of the global corner id /is/ the number of the
+ * corner in the local CNS, and we use this fact.
+ *
+ *     The algorithm begins by determining the shared corners between the
+ * Blocks:
+ *     result: {4,1,10,7}
+ *
+ *     The next step is to determine the local ids of these shared global ids:
+ *     Block1 result: {2,3,6,7}
+ *     Block2 result: {2,0,6,4}
+ *
+ *     The next step is to convert these ids into their binary representation:
+ *     Block1 result: {010,011,110,111}
+ *     Block2 result: {010,000,110,100}
+ *
+ *     Note that we can now read off the shared face relative to each Block
+ * easily:
+ *     Block1 result: Upper eta (All binary Block1 ids have a 1 in the second
+ * position)
+ *     Block2 result: Lower xi (All binary Block2 ids have a 0 in the 3rd
+ * position)
+ *
+ *     Now we know that Direction<3>::upperEta() in Block1 corresponds to
+ * Direction<3>::lowerXi().opposite()
+ * in Block2.
+ *     ".flipped()" is a result of the Directions to a Block face being
+ * anti-parallel
+ *     because each Block lies on the opposite side of the shared face.
+ *
+ *    The remaining two correspondences are given by the Alignment of the shared
+ * face.
+ *    It is useful to know the following information:
+ *    In the Local CNS, if an edge lies along the xi direction, if one takes the
+ * two corners making up that edge and takes the
+ *    difference of their ids, one always gets the result +/- 1.
+ *    Similarly, if the edge lies in the eta direction, the result will be +/-
+ * 2.
+ *    Finally, if the edge lies in the zeta direction, the result will be +/- 4.
+ *    We use this information to determine the Alignment of the shared face:
+ *
+ *    Block1: 3-2=1  => This edge is in the +xi direction.
+ *    Block2: 0-2=-2 => This edge is in the -eta direction.
+ *    Then, +xi in Block1 corresponds to -eta in Block2.
+ *
+ *    Block1: 6-2=4 => This edge is in the +zeta direction.
+ *    Block2: 6-2=4 => This edge is in the +zeta direction.
+ *    Then, +zeta in Block1 corresponds to +zeta in Block2.
+ *
+ *    The corresponding directions in each Block have now been deduced.
+ *
+ *    To confirm, we can use the other ids as well and arrive at the same
+ * result:
+ *    Block1: 7-6-1  => +xi
+ *    Block2: 4-6=-2 => -eta
+ *
+ *    Block1: 7-3=4  => +zeta
+ *    Block2: 4-0=4  => +zeta
+ *
+ *  ----------------------------------------------------------------------
+ *
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <numeric>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+
+template <size_t>
+class BlockNeighbor;
+
+template <size_t>
+class Direction;
+
+template <size_t, size_t>
+class EmbeddingMap;
+
+template <size_t VolumeDim>
+struct DomainHelpers {
+  DomainHelpers();
+
+  /*!
```

Comment:
Since we're allowing /// doxy comments I'd use those here so that these only take one line instead of three.